File: frontend/src/utils/skipDeductionNote.js
export const SKIP_DEDUCTION_NOTE = '未休息不扣减'

export function applySkipDeductionNote(note, skip) {
  const parts = String(note || '')
    .split('、')
    .map((p) => p.trim())
    .filter(Boolean)
    .filter((p) => p !== SKIP_DEDUCTION_NOTE)
  if (skip) parts.push(SKIP_DEDUCTION_NOTE)
  return parts.join('、')
}

File: backend/app/services/skip_deduction_note.py
SKIP_DEDUCTION_NOTE = "未休息不扣减"


def apply_skip_deduction_note(note: str | None, skip: bool) -> str | None:
    parts = [p.strip() for p in str(note or "").split("、") if p.strip()]
    parts = [p for p in parts if p != SKIP_DEDUCTION_NOTE]
    if skip:
        parts.append(SKIP_DEDUCTION_NOTE)
    if not parts:
        return None
    return "、".join(parts)
